docs/proposals/client-settings.md

```diff
@@ -0,0 +1,395 @@
+
+**3: Gateway and HTTPRoute Attachment**
+
+![attach-3](/docs/images/client-settings/attach-3.png)
+
+When a `ClientSettingsPolicy` is attached to a Gateway and one or more of the HTTPRoutes that are attached to that Gateway, the effective policy is calculated by accepting the "lowest" default configured.
+Let's examine each HTTPRoute's effective policy:
+
+- `baz`: This HTTPRoute has a `ClientSettingsPolicy` attached to it that sets both the max body size and the body timeout. It's also affected by the `ClientSettingsPolicy` attached to the Gateway, which also sets these two values. Since HTTPRoutes are a "lower" object than Gateways in the inheritance hierarchy, the defaults set by the policy attached to the `baz` route win.
+- `bar`: This HTTPRoute does not have a policy attached to it but is affected by the policy attached to the Gateway. As a result, it inherits the settings from the Gateway's policy.
+- `foo`: This HTTPRoute is similar to the `baz` route because it is affected by two policies: the policy attached to the Gateway and the policy attached to itself. However, since the policy attached to itself only sets the value for max body size, the `foo` route inherits the value for body timeout from the Gateway policy.
+
+For more information on how to calculate effective policies, see the [hierarchy](https://gateway-api.sigs.k8s.io/geps/gep-713/#hierarchy) and [merging](https://gateway-api.sigs.k8s.io/geps/gep-713/#merging-into-existing-spec-fields) sections in the Policy and Metaresources GEP.
+
+### NGINX Inheritance Behavior
+
+The client settings NGINX directives are available in three NGINX contexts: http, server, and location. Similar to Gateway API resources, there is a hierarchy among these contexts:
+
+```mermaid
+graph TD
+    http --> server --> location
+```
+
+In general, NGINX directives inherit downwards only. The location context inherits values set above it in the server and http contexts, and the server context inherits values set in the http context. Additionally, directives specified in a lower context will replace those specified in a higher context. So, a directive set in the location context will replace those set in the server and http contexts.
+When a request is rewritten internally from one location to another, only directives in the second location apply. In other words, a location cannot inherit directives from another location.
+
+There are some nuances to the inheritance behavior when dealing with array directives and action directives (see [NGINX Inheritance Behavior](https://blog.martinfjordvald.com/understanding-the-nginx-configuration-inheritance-model/) for more information), but the directives included in this policy follow this top-down inheritance pattern.
+
+### Mapping Gateway API Resources to NGINX Contexts
```

Comment:
would it make sense to put this section into a separate doc? 
cause this content is useful in other contexts (pun intended) like explaining how NGF generates NGINX configuration and probably for other future policies too.